templates/Template.NEP5.CSharp/NEP5.Crowdsale.cs

```diff
@@ -0,0 +1,74 @@
+using Neo.SmartContract.Framework;
+using Neo.SmartContract.Framework.Services.Neo;
+using Neo.SmartContract.Framework.Services.System;
+using System;
+using System.Numerics;
+
+namespace Template.NEP5.CSharp
+{
+    public partial class NEP5 : SmartContract
+    {
+        private static BigInteger GetTransactionAmount(object state)
+        {
+            var notification = (object[])state;
+            // Checks notification format
+            if (notification.Length != 4) return 0;
```

Comment:
<4? if in the future we add more entries we should add them at the end, and the contract could works.
```suggestion
            if (notification.Length < 4) return 0;
```

But this kind of things must be versioning in the abi, in order to don't affect other contracts. so it's good to me like now as well.

Reply:
At the moment I'd prefer to leave it with the current notifications format, is it ok?